src/features/moderation/logs/ModlogItem.tsx

```diff
@@ -14,80 +14,154 @@ import removeComment from "./types/removeComment";
 import removeCommunity from "./types/removeCommunity";
 import removePost from "./types/removePost";
 import transferCommunity from "./types/transferCommunity";
-import { IonItem } from "@ionic/react";
+import { IonIcon, IonItem } from "@ionic/react";
 import { maxWidthCss } from "../../shared/AppContent";
 import Ago from "../../labels/Ago";
 import { useBuildGeneralBrowseLink } from "../../../helpers/routes";
+import {
+  AdminPurgeCommentView,
+  AdminPurgeCommunityView,
+  AdminPurgePersonView,
+  AdminPurgePostView,
+  ModAddCommunityView,
+  ModAddView,
+  ModBanFromCommunityView,
+  ModBanView,
+  ModFeaturePostView,
+  ModHideCommunityView,
+  ModLockPostView,
+  ModRemoveCommentView,
+  ModRemoveCommunityView,
+  ModRemovePostView,
+  ModTransferCommunityView,
+} from "lemmy-js-client";
+import ModlogItemMoreActions from "./ModlogItemMoreActions";
+import {
+  ModeratorRole,
+  getModColor,
+  getModIcon,
+  getModName,
+} from "../useCanModerate";
+import useIsAdmin from "../useIsAdmin";
+import { timerOutline } from "ionicons/icons";
 import { styled } from "@linaria/react";
 
-const Contents = styled.div`
+const Container = styled.div`
+  display: flex;
+  gap: 1rem;
+
+  ${maxWidthCss}
+
+  padding: 0.5rem 0;
+
   font-size: 0.875em;
-  width: 100%;
 
+  strong {
+    font-weight: 500;
+  }
+`;
+
+const StartContent = styled.div`
   display: flex;
   flex-direction: column;
-  gap: 6px;
-  padding: 8px 0;
+  gap: 1rem;
+`;
+
+const Content = styled.div`
+  flex: 1;
 `;
 
-const TitleLine = styled.div`
+const Header = styled.div`
   display: flex;
   justify-content: space-between;
 
   aside {
     color: var(--ion-color-medium);
+    margin-left: auto;
+
+    display: flex;
+    align-items: center;
+    gap: 0.5rem;
   }
 `;
 
-const Title = styled.div``;
-const Message = styled.div`
+const Body = styled.div`
   color: var(--ion-color-medium);
+  padding: 0.5rem 0;
+`;
+
+const Footer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+
+  aside {
+    color: var(--ion-color-medium);
+    display: flex;
+    align-items: center;
+    gap: 6px;
+  }
+`;
+
+const Title = styled.div``;
+const Reason = styled.div``;
+const By = styled.div<{ color: string }>`
+  display: flex;
+  justify-content: right;
+  align-items: center;
+  gap: 6px;
+  padding: 0.5rem 0;
+  color: ${({ color }) => `var(--ion-color-${color}-shade)`};
 `;
 
 interface ModLogItemProps {
   item: ModlogItemType;
 }
 
 export interface LogEntryData {
+  icon: string;
   title: string;
   when: string;
   message?: string;
+  reason?: string;
+  expires?: string;
   by?: string;
+  role?: ModeratorRole;
   link?: string;
 }
 
 function renderModlogData(item: ModlogItemType): LogEntryData {
   switch (true) {
     case "mod_remove_comment" in item:
-      return removeComment(item);
+      return removeComment(item as ModRemoveCommentView);
```

Comment:
Why were these `as` added? Typescript should be able to infer this

Reply:
I made this months ago, it's possible that VS Code was giving me warnings, but perhaps that's changed. I'll review.

Reply:
Yep, looks like it's no longer necessary. Removed.